lib/diaspora_federation/entity.rb

```diff
@@ -264,21 +315,52 @@ def simple_node(doc, name, value)
       end
     end
 
-    # @param [Nokogiri::XML::Element] root_node xml nodes
+    def json_data
+      enriched_properties.map {|key, value|
+        type = self.class.class_props[key]
+
+        if type.instance_of?(Class) && value.respond_to?(:to_json_hash)
+          entity_data = value.to_json_hash[:entity_data]
+          [key, entity_data] unless entity_data.nil?
+        else
+          [key, value]
+        end
```

Comment:
Hmm, I also thought a bit more about this. So we are still relaying on that somebody uses `ActiveSupport` with the resulting hash?

So we handle single nested entities, because we have the `nil?` check anyway. And I think it doesn't hurt when we also call `.to_json` explicitly for nested arrays here, because then we have a clean hash returned here.